backend_new/src/services/ami-chart.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { Prisma } from '@prisma/client';
+import { AmiChart } from '../dtos/ami-charts/ami-chart-get.dto';
+import { AmiChartCreate } from '../dtos/ami-charts/ami-chart-create.dto';
+import { AmiChartUpdate } from '../dtos/ami-charts/ami-chart-update.dto';
+import { AmiChartQueryParams } from '../dtos/ami-charts/ami-chart-query-params.dto';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+
+/*
+  this is the service for ami charts
+  it handles all the backend's business logic for reading/writing/deleting ami chart data
+*/
+
+const view: Prisma.AmiChartInclude = {
+  jurisdictions: true,
+};
+
+@Injectable()
+export class AmiChartService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of ami charts given the params passed in
+  */
+  async list(params: AmiChartQueryParams) {
+    const rawAmiCharts = await this.prisma.amiChart.findMany({
+      include: view,
+      where: this.buildWhereClause(params),
+    });
+    return mapTo(AmiChart, rawAmiCharts);
+  }
+
+  /*
+    this helps build the where clause for the list()
+  */
+  buildWhereClause(params: AmiChartQueryParams): Prisma.AmiChartWhereInput {
+    const filters: Prisma.AmiChartWhereInput[] = [];
+
+    if (params) {
+      if ('jurisdictionName' in params && params.jurisdictionName) {
+        filters.push({
+          jurisdictions: {
+            name: params.jurisdictionName,
+          },
+        });
+      } else if ('jurisdictionId' in params && params.jurisdictionId) {
+        filters.push({
+          jurisdictions: {
+            id: params.jurisdictionId,
+          },
+        });
+      }
+    }
+
+    return {
+      AND: filters,
+    };
+  }
+
+  /*
+    this will return 1 ami chart or error
+  */
+  async findOne(amiChartId: string) {
+    const amiChartRaw = await this.prisma.amiChart.findFirst({
+      include: view,
+      where: {
+        id: {
+          equals: amiChartId,
+        },
+      },
+    });
+
+    if (!amiChartRaw) {
+      throw new NotFoundException();
+    }
+
+    return mapTo(AmiChart, amiChartRaw);
+  }
+
+  /*
+    this will create an ami chart
+  */
+  async create(incomingData: AmiChartCreate) {
+    const rawResult = await this.prisma.amiChart.create({
+      data: {
+        ...incomingData,
+        items: JSON.stringify(incomingData.items),
```

Comment:
I noticed the items field now escapes all of the quotes. It looks intentional based on this code, but do we have any concerns with this difference?
e.g this is in bloom_prisma
`"[{\"percentOfAmi\":50,\"householdSize\":3,\"income\":65850},`
vs this in the bloom db
`[{"income":65850,"percentOfAmi":50,"householdSize":3},`

Reply:
I didn't catch this! I added a transform function to the DTO so if it happens to be stringified it is no longer stringified